src/zarr/store/local.py

```diff
@@ -193,14 +193,10 @@ async def list_prefix(self, prefix: str) -> AsyncGenerator[str, None]:
         -------
         AsyncGenerator[str, None]
         """
-        for p in (self.root / prefix).rglob("*"):
-            if p.is_file():
-                yield str(p)
```

Comment:
Were we getting duplicates?

Reply:
i think we were. this code path was not tested until this PR